backend/src/controllers/items-controller.ts

```diff
@@ -0,0 +1,136 @@
+import { Request, Response } from 'express';
+import { validationResult } from 'express-validator';
+import Item from '../models/item';
+import User from '../models/user'
+import InventoryItem from "../models/inventory-item";
+import mongoose from 'mongoose';
+import {ObjectId} from "mongodb";
+
+//Creates an item
+export const createItem = async (req: Request, res: Response) => {
+    //Validation check
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        console.log("errors", errors);
+        return res.status(400).json({errors: errors.array()});
+    }
+    //Duplicated item check
+    const dupeItem = Item.findOne({name: req.body.name})
+    if (!dupeItem) {
+        return res.status(409).json({error: "An item with the same name already exists."});
+    }
+    //Checks if the ID matches the owner of the item
+    const itemOwnerId = req.body._id;
+    const itemOwnerExistence = await User.findOne({_id: itemOwnerId});
+    if (!itemOwnerExistence) {
+        return res.status(404).json({error: "This id does not match the owner of the item."});
+    }
+    const name = req.body.name as string;
+    const description = req.body.description as string;
+    const pickupLocation = req.body.pickupLocation as string;
+    const pickupTime = req.body.pickupTime as string;
+
+
+
+    //Interface for a new item
+    const item = new Item({
+        name: name,
+        description: description,
+        pickupLocation: pickupLocation,
+        pickupTime: pickupTime,
+        itemOwnerId: itemOwnerExistence._id
+    });
+    //Creates a transaction to create inventory-items
+    //Checks if there's missing sizes
+    const transaction = await mongoose.startSession();
+    transaction.startTransaction();
+    try {
+        const savedItem = await item.save();
+        const sizeList = ['XXS', 'XS', 'S', 'M', 'L', 'XL', 'XXL'];
+        for (const size of sizeList) {
+            const invItem = new InventoryItem({
```

Comment:
Creation of `inventory-item` triggers this error:

![image](https://github.com/miguelaenlle/merch-order-request-form/assets/66172204/65c78f2b-e4e6-409e-94c7-cb203d725a15)

Check the `inventory-item` schema:
- `item` field should be `itemId`
- `amount` field should default to zero
- In the `request` `body`, retrieve a `price` from the user. Then, set all inventory-item sizes' prices to that price.